linode/linode_instance_helpers.go

```diff
@@ -702,25 +712,53 @@ func changeInstanceType(client *linodego.Client, instance *linodego.Instance, ta
 			return fmt.Errorf("Error waiting for instance %d readiness: %s", instance.ID, err)
 		}
 	}
+	// We have to wait through the resize process because if we issue jobs before the complete process is complete, the API
+	// Will raise an error.
 
-	if err := client.ResizeInstance(context.Background(), instance.ID, targetType); err != nil {
+	// Issue the resize job
+	if err := client.ResizeInstance(context.Background(), instance.ID, resizeOpts); err != nil {
 		return fmt.Errorf("Error resizing instance %d: %s", instance.ID, err)
 	}
 
-	_, err := client.WaitForEventFinished(context.Background(), instance.ID, linodego.EntityLinode, linodego.ActionLinodeResize, *instance.Created, int(d.Timeout(schema.TimeoutUpdate).Seconds()))
-	if err != nil {
-		return fmt.Errorf("Error waiting for instance %d to finish resizing: %s", instance.ID, err)
+	// wait for instance to begin resizing after issuing resize job
```

Comment:
Waiting for a transitional status is risky because the event may have already been transitioned out by the time the code gets here, or it may have transitioned from pre-resizing to post-resizing in between polling loops.  This is why EventForEventFinished was used.

Reply:
I see.. 

I swapped out WaitForEventFinished in favor of WaitForInstanceStatus because there are situations -- namely when resizing a disk after resizing an instance -- where the instance resize event will have completed, but the issuing additional jobs before the Linode has been returned to its original status results in the API returning an error because the Linode is still "busy"..

I understand your concern about waiting for the transitional "InstanceResizing" status here. If we were to nix waiting for the transitional "resize" status, though, and only wait for "Running" or "Offline," then there are situations where this will simply move ahead while the instance is in state "Running" or "Offline", while waiting for its turn in the migration queue. I'm not sure that I see a good alternative but I welcome suggestions.

Reply:
I nixed the code where it waits for the InstanceResizing status. Like you suggested, I'm using WaitForEventFinished, and then waiting for the instance to return to its original state. This seems to work and I agree that it will avoid the possibility of missing the transitional status.